architectureDecisionRecords/0009HerokuDeployment.md

```diff
@@ -0,0 +1,56 @@
+# Title
+
+CI/CD Process
+
+## Context
+
+As was [previously decided](0005CloudServiceProvider.md), Heroku is the cloud service provider for UCP. In order to follow best practices, all services deployed to Heroku will follow a standard CI/CD process. Automating the deployment process will prevent human-caused errors, allow us to deploy small code changes frequently, and reduce overall risk to the project.
+
+Within [Heroku pipelines](https://devcenter.heroku.com/articles/pipelines), there are 2 options for deployments: using Heroku to automatically deploy upon merging to the main branch, or using github actions to manually deploy a pull request.
+
+For a standard Dev/Staging/Production environment set up, that leaves 2 options for deployments:
+
+1. All deployments within Heroku
+
+ * Dev environment: Heroku automatically deploys upon PR creation/update
+ * Staging environment: Heroku automatically deploys upon PR merge into main
+ * Production environment: Developer deploys code from Staging to Production by clicking the "Promote to production" button in Heroku
```

Comment:
Is it not possible to have it automatically deploy to both staging and production on merge to main?

Reply:
The only reason I could see to not want an automatic deploy to prod is for testing in staging. Can we have tests run in staging, and then successful tests trigger a prod deployment?